nozzle/api_client.go

```diff
@@ -0,0 +1,103 @@
+package nozzle
+
+import (
+	"log"
+	"time"
+
+	cfclient "github.com/cloudfoundry-community/go-cfclient"
+	cache "github.com/patrickmn/go-cache"
+)
+
+// APIClient wrapper for Cloud Foundry Client
+type APIClient struct {
+	clientConfig *cfclient.Config
+	client       *cfclient.Client
+	appCache     *cache.Cache
+	logger       *log.Logger
+}
+
+// AppInfo holds Cloud Foundry applications information
+type AppInfo struct {
+	Name  string
+	Space string
+	Org   string
+}
+
+func newAppInfo(app cfclient.App) *AppInfo {
+	space, _ := app.Space()
```

Comment:
Handle errors. If error in retrieving space or org let's default to `not_found` or such.